python/graphstorm/model/edge_gnn.py

```diff
@@ -187,41 +187,48 @@ def edge_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
     model.eval()
 
     len_dataloader = max_num_batch = len(loader)
-    tensor = th.tensor([len_dataloader], device=device)
+    num_batch = th.tensor([len_dataloader], device=device)
     if is_distributed():
-        th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
-        max_num_batch = tensor[0]
-
+        th.distributed.all_reduce(num_batch, op=th.distributed.ReduceOp.MAX)
+        max_num_batch = num_batch[0]
     dataloader_iter = iter(loader)
 
     with th.no_grad():
         # WholeGraph does not support imbalanced batch numbers across processes/trainers
         # TODO (IN): Fix dataloader to have the same number of minibatches
         for iter_l in range(max_num_batch):
             iter_start = time.time()
-            tmp_keys = []
-            blocks = None
+            tmp_node_keys = tmp_edge_keys = []
+            blocks = target_edge_graph = None
             input_nodes = {}
+            input_edges = {} if data.decoder_edge_feat is not None else None
```

Comment:
```suggestion
            input_edges = {}
```

Reply:
Input_edges is only used when data.decoder_edge_feat is not None, so we can just init it as an empty dict at the beginning.